### ISSUE TITLE
Simple Jobs scheduled at same time won't all execute the jobclass

### ISSUE THREAD

**Issue by algernon28** [quartz-scheduler/quartz#923]
Issue: When I schedule two or more Jobs with Simple Triggers at the same time, both are acquired, both fired (they are removed from the database), but only one will execute the class.

I am Using SpringBoot 3.0
Quartz 2.3.2
The Job extends QuartzJobBean, I tried with and without @DisallowConcurrentExecution.
I am getting crazy around this issue, I would expect Quartz executing my jobclass both JobDetails.
The triggers are different (Mine are all one-timers). 

This is my Quartz Configuration

  quartz:
    scheduler-name: QuartzScheduler
    auto-startup: true
    wait-for-jobs-to-complete-on-shutdown: false
    overwrite-existing-jobs: false
    startup-delay: 0
    job-store-type: JDBC
    jdbc:
      #Use ALWAYS to generate the schema, then switch to NEVER or it will truncate all the tables every time the app runs
      #initialize-schema: ALWAYS
      initialize-schema: NEVER
      platform: postgres
    properties:
      org:
        quartz:
          scheduler:
            #Specifies the name of the Quartz scheduler instance.
            #This is used to differentiate between multiple Quartz scheduler instances that may be running in the same JVM.
            instanceName: "MyScheduler"
          jobStore:
            class: org.quartz.impl.jdbcjobstore.JobStoreTX
            driverDelegateClass: org.quartz.impl.jdbcjobstore.PostgreSQLDelegate
            dataSource: schedulerDS  #Datasource logical name
            tablePrefix: QRTZ_
            misfireThreshold: 60000 # default. Trigger is considered misfired after 60 seconds.
          dataSource:
            schedulerDS:
              driver: org.postgresql.Driver
              URL: jdbc:postgresql://127.0.0.1:5432/mydb
              schema: utopia
              user: jobscheduler
              password: jobscheduler
              provider: hikaricp

![image](https://github.com/quartz-scheduler/quartz/assets/11425650/e1cd6cb9-ff65-41d9-bb5b-8443a3a1b4e0)
![image](https://github.com/quartz-scheduler/quartz/assets/11425650/ff6f33c2-0e96-40ea-ab34-7eef496d8994)
![image](https://github.com/quartz-scheduler/quartz/assets/11425650/5217d5eb-8e97-4674-836a-e6e558b81e89)

Below my audit table. The highlighted trigger fired but the Job class execute() was not invoked.
![image](https://github.com/quartz-scheduler/quartz/assets/11425650/948626a0-aa57-4139-9d72-e1584aa50842)

**Comment by jhouserizer** [comment]
The only thing that I can think to ask:  are you REALLY sure that the execute method on the job is not getting invoked?  Perhaps add a log or print message as the very first thing it does (if you can't run in a debugger with a breakpoint there) - to be fully sure.
Perhaps also look at quartz logs etc. to see if quartz is failing to instantiate the given job (and then troubleshoot why).

**Comment by algernon28** [comment]
I am sure.
I have plenty of logs.
Now I have increased the threadpool size and the problem seems gone, but I am still concerned because I would expect to see jobs blocked until there are threads available.

With 10 threads I was still having problems scheduling 5 jobs at same time.
All of them with the same jobclass and short lived.
2 invoked an external URL and 3 only printed logs.
I can see that Quartz acquires all of them (triggers in ACQUIRED state), fires them and then removes them from the table.
This is expected, what is not expected is some jobs not being invoked.

I cannot determine a pattern, which one gets dropped seems to be random.
I also implemented listeners or all kind, I can confirm that 1 or more jobs (depending on the threadpool size) is not invoked.
Just increasing the size doesn't seem a permanent solution if jobs are just dropped.

I thought jobs would be somewhat queued if there aren't threads available, but that doesn't seem to be the case

**Comment by algernon28** [comment]
Someone could try to see if they can replicate it?
Schedule 5 simple jobs at the same time (at the millisecond).
Java 17
Springboot 3.0

The jobclass is a QuartzJobBean, invoking a spring bean via DI to write the audit tables.

I set the batch acquisition ahead time to 1s, and tried to bump it up to 30s.

Try default threadpool and batch acquisition size (the behaviour changed when I bumped up these, but it didn't completely disappear until the values became very high, like 20)

I watched the dB tables, the logs, I implemented listeners, I debugged.
Sometimes the jobclass executeInternal () is just never invoked.
